components/proxy/src/main/java/com/hotels/styx/routing/handlers/StandardHttpPipeline.java

```diff
@@ -61,16 +63,18 @@ public StandardHttpPipeline(List<HttpInterceptor> interceptors, HttpHandler hand
         private final int index;
         private final HttpHandler client;
         private final HttpInterceptor.Context context;
+        private final RequestTracker requestTracker;
 
-        HttpInterceptorChain(List<HttpInterceptor> interceptors, int index, HttpHandler client, HttpInterceptor.Context context) {
+        HttpInterceptorChain(List<HttpInterceptor> interceptors, int index, HttpHandler client, HttpInterceptor.Context context, RequestTracker requestTracker) {
             this.interceptors = interceptors;
             this.index = index;
             this.client = client;
             this.context = context;
+            this.requestTracker = requestTracker;
```

Comment:
```suggestion
            this.requestTracker = requireNonNull(requestTracker);
```
The other fields should probably have this too (besides `index`).